apigw-lambda-dynamodb-sam-go/README.md

```diff
@@ -0,0 +1,74 @@
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+
+    
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+   ```bash
+   git clone https://github.com/aws-samples/serverless-patterns
+   ```
+2. Change directory to the pattern directory:
+   ```bash
+   cd apigw-lambda-dynamodb-go-sam
+   ```
+3. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+   ```bash
+   sam deploy --guided
+   ```
+4. During the prompts:
+
+   - Enter a stack name
+   - Enter the desired AWS Region
+   - Allow SAM CLI to create IAM roles with the required permissions.
+
+   Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+
+   When asked "`ContentGenerationLambdaFunction` has no authentication. Is this okay? [y/N]", answer explicitly with y for the purposes of this sample application. As a result, anyone will be able to call this example REST API without any form of authentication.
+
+   For production applications, you should [enable authentication for the API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-to-api.html) using one of several available options and [follow the API Gateway security best practices](https://docs.aws.amazon.com/apigateway/latest/developerguide/security-best-practices.html).
+
+5. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for next step as well as testing.
+
+
+## How it works
+
+This SAM project uses Amazon DynamoDB to store user data in a single table. This is exposed through a serverless REST API.
+
+Here's a breakdown of the steps:
+
+1. **Amazon API Gateway**: Receives the HTTP POST request containing the prompt.
+
+2. **AWS Lambda**: Triggered by the API Gateway, this function grabs the request body and uses the User data to create a new item in DynamoDB.
+
+4. **Response**: Lambda processes the request and returns a success or error as a response to the API Gateway.
+
+## Testing
+
+Test the deployed content generation API by providing a prompt. You can use [curl](https://curl.se/) to send a HTTP POST request to the API. Make sure to replace `API_Endpoint` with the one from your `sam deploy --guided` output:
+
+```bash
+curl -d '{"id": "12345", "first_name":"John", "last_name":"Doe"}' -H 'Content-Type: application/json'  <API_Endpoint>
```

Comment:
This would most likely return a 5XX error as the function go code was not compiled to publish binary